shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -94,6 +94,58 @@
  */
 @Keep
 public class FlutterJNI {
+  public static class FlutterJNILoader {
```

Comment:
Making all the static private SG. But rather than making this class 2 tiered which is a bit more confusing, why not just let classes that want to mock FlutterJNI just inject a FlutterJNI mock directly so all methods can be mocked?

Reply:
It limits the scope of what `FlutterLoader` gets, making it a little easier to quickly see exactly what it expects to use  and how these things are related.

Reply:
IOW, writing tests for this won't leave me wondering which of the many methods on FlutterJNI I'm supposed to mock.

Reply:
Jason might have the historical background. I believe we made all the methods in the FlutterJNI class grouped in one class instead of being sliced and diced per consumer based on least privilege principle for a reason (probably to have a single place to look at the interop schema). If so, I strongly believe we should have a single pattern in the codebase to be consistent. Either keep everything in one class (and remove statics which I support) or break it all up and parcel views of the JNI to its consumer based on what the consumer needs. 

Otherwise, it becomes arbitrary why I could mock the FlutterLoader's JNI loader but on some JNI calls, it still crashes. 

I think the cleanest way for your intended change is still to make all the statics normal instance methods for the public APIs and inject the whole FlutterJNI in tests.

Reply:
The actual native methods are still on Flutter JNI, which still keeps it easier for us to write the JNI bindings. 

I'm not really clear on why we would want to keep all of the Java facing methods on the same class. Flutter JNI will stick around as this sort of God-object, and everyone you give it to will be able to call all sorts of native methods that may not make sense. 

I don't work on this code very much, so I'm not going to put up a huge fight about this. If, seeing what I'm saying, you still believe these methods should all be exposed on the Java side via the same one object, I can easily refactor this PR to do that - although if I were working on this more I would more strongly object :)

Reply:
Talked offline. I'll move out this part of the change here, we can track what to do with FlutterJNI overall in https://github.com/flutter/flutter/issues/66131

I've also made the static methods public again, but marked them as deprecated with pointers to the instance methods.

Reply:
Ping me when it's ready again

Reply:
It should be ready now :) The test failure is an infra flake.